exec/vector/src/main/java/org/apache/drill/exec/record/metadata/MapBuilder.java

```diff
@@ -144,6 +144,14 @@ public RepeatedListBuilder addRepeatedList(String name) {
     return tupleBuilder.addRepeatedList(this, name);
   }
 
+  public DictBuilder addDict(String name) {
+    return tupleBuilder.addDict(this, name);
+  }
```

Comment:
Maybe `addDict(String name, MinorType keyType)`. Works because the key must be a scalar and is required. Then the `DictBuilder` simply builds the value type:

```
scalarValue(MinorType type);
MapBuilder mapValue();
MapBuilder mapArrayValue();
UnionBuilder unionValue();
Foo resumeFoo();
```
Can the value be nullable? (The comment in `DictVector` doesn't say yes or no.) If nullable is allowed, then only the scalar can be nullable, so maybe add:

```
nullableValue(MinorType type)
```

Reply:
Good point. Yes, `value` can be nullable (added explicit comment to `DictVector`).